tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -65,53 +64,57 @@ internal static TReturn EndInvocationAsync<TReturn>(TReturn returnValue, Excepti
             var json = SerializeObject(returnValue);
             scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null, json);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null, json);
             return returnValue;
         }
 
         internal static CallTargetReturn EndInvocationVoid(Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             scope?.Dispose();
```

Comment:
To make the code a little cleaner, can you move the `scope?.Dispose()` call after the `NotifyExtensionEnd` call? Now you're using the scope in that method and style-wise we should dispose the scope after it's finished being used

Reply:
✅